disco_aws_automation/resource_helper.py

```diff
@@ -71,10 +71,8 @@ def keep_trying(max_time, fun, *args, **kwargs):
                 logger.exception("Failed to run %s.", fun)
             if time.time() > expire_time:
                 raise
-            time.sleep(curr_delay)
-            delay_register = last_delay
-            last_delay = curr_delay
-            curr_delay = min(curr_delay + delay_register, MAX_POLL_INTERVAL)
+            cycle += 1
+            backoff(cycle)
```

Comment:
Suggestion: why not having the incrementation of the cycle in the back of function and return that value